### PR TITLE
Remove Deprecated Methods:

### LINKED ISSUE #4825: Removed deprecated methods

### PR DESCRIPTION
**Pull request by takameyer**
## What, How & Why?
* The following deprecated methods have been removed: * `Realm#automaticSyncConfiguration` * `Realm.Credentials#google` with `authCode` parameter (use `authObject`)
* `Realm#writeCopyTo` now only accepts an output Realm configuration as a parameter.

This closes #4825 and #4831

## ☑️ ToDos

* [x] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 📝 Update `COMPATIBILITY.md`
* [x] 🚦 Tests
* [ ] 🔀 Executed flexible sync tests locally if modifying flexible sync
* [ ] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)
* [ ] 📱 Check the React Native/other sample apps work if necessary
* [ ] 📝 Public documentation PR created or is not necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

*If this PR adds or changes public API's:*
* [x] typescript definitions file is updated
* [x] jsdoc files updated
* [ ] Chrome debug API is updated if API is available on React Native

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,9 +5,13 @@ Based on Realm JS v10.21.1: See changelog below for details on enhancements and
 
 <<<<<<< HEAD
```
Comment (kneth) [inline CHANGELOG.md:6]: Looks wrong :-)

**Review by kneth — APPROVED** [review]
Minor comment only - ship it!

File: docs/realm.js [hunk 2]
```diff
@@ -308,27 +297,17 @@ class Realm {
...
    *
    * Note that if this method is called from within a write transaction, the current data is written,
    * not the data from the point when the previous write transaction was committed.
-   * @param {string|Realm~Configuration} pathOrConfig a α) path to save the Realm to, OR β) {@link Realm~Configuration | Configuration} that describes the output realm.
-   * @param {ArrayBuffer|ArrayBufferView} [encryptionKey] - Optional 64-byte encryption key to encrypt the new file with.  Must not be present when
-   * β) a {@link Realm~Configuration | Configuration} is given as first parameter, in which case encryptionKey can be set in {@link Realm~Configuration#encryptionKey}.
+   * @param {@link Realm~Configuration | Configuration} that describes the output Realm.
    */
-  writeCopyTo(pathOrConfig, encryptionKey) {}
+  writeCopyTo(pathOrConfig) {}
```
Comment (kraenhansen) [inline docs/realm.js:310]:
```suggestion
  writeCopyTo(config) {}
```

File: src/js_app_credentials.hpp [hunk 3]
```diff
@@ -126,13 +127,9 @@ void CredentialsClass<T>::google(ContextType ctx, ObjectType this_object, Argume
...
-            std::string token = Value::validated_to_string(ctx, arguments[0]);
-            if (token.substr(0, 2) == "4/") {
-                return app::AppCredentials::google(app::AuthCode(token));
-            }
-            else if (token.substr(0, 2) == "ey") {
-                return app::AppCredentials::google(app::IdToken(token));
-            }
+            log_to_console<T>(ctx,
+                              "`google(<tokenString>)` has been deprecated.  Please use `google(<authCodeObject>).",
+                              JSLogFunction::Error);
```
Comment (kraenhansen) [inline src/js_app_credentials.hpp:132]: The other deprecations (that just got removed), used warnings instead of error:

```suggestion
                              JSLogFunction::Warning));
```

Comment (kraenhansen) [reply src/js_app_credentials.hpp:132]: I imagine this needs a change in `realm-web` too?

File: tests/js/realm-tests.js [hunk 4]
```diff
@@ -2033,56 +2033,6 @@ module.exports = {
     realm.close();
   },
 
-  testWriteCopyTo: function () {
```
Comment (kraenhansen) [inline tests/js/realm-tests.js:2036]: Should this really be removed in its entirety? That leaves the feature completely untested, doesn't it?

File: tests/js/session-tests.js [hunk 5]
```diff
@@ -1182,11 +1182,18 @@ module.exports = {
...
+        user: user1,
+        partitionValue: partition,
+        _sessionStopPolicy: "immediately", // Make it safe to delete files after realm.close()
+      },
+      schema: [schemas.PersonForSync, schemas.DogForSync],
+      path: `${realm1Path}copy1.realm`,
+    };
 
-    // log out the user that created the realm
-    await user1.logOut();
```
Comment (kraenhansen) [inline tests/js/session-tests.js:1189]: Is this not an intended part of the test? Why did this logout get removed?

File: src/js_realm.hpp [hunk 6]
```diff
@@ -1734,14 +1670,15 @@ void RealmClass<T>::writeCopyTo(ContextType ctx, ObjectType this_object, Argumen
     realm::Realm::Config config;
     if (args.count == 0) {
         throw std::invalid_argument(
-            "`writeCopyTo` requires <output configuration> or <path, [encryptionKey]> parameters");
+            "`writeCopyTo` requires <output configuration> as a parameter. See documentation for details.");
```
Comment (kraenhansen) [inline src/js_realm.hpp:1673]:
```suggestion
            "`writeCopyTo` requires <output configuration> as a parameter.

Comment (kraenhansen) [reply src/js_realm.hpp:1673]: I'd suggest leaving out the method name from the error message, as that will show from the callstack and is redundant as such.

File: types/app.d.ts [hunk 7]
```diff
@@ -207,14 +207,6 @@ declare namespace Realm {
      */
     static jwt(token: string): Credentials<Credentials.JWTPayload>;
 
-    /**
-     * Factory for `Credentials` which authenticate using the [Google Provider](https://docs.mongodb.com/realm/authentication/google/).
-     *
-     * @param authCode The auth code returned from Google.
-     * @returns A `Credentials` object for logging in using `app.logIn`.
-     */
-    static google(authCodeOrIdToken: string): Credentials<Credentials.GooglePayload>;
```
Comment (kraenhansen) [inline types/app.d.ts:216]: Why is this removed? Wasn't it just supposed to take an object as argument?

File: tests/js/realm-tests.js (hunk 4 above)
Comment (takameyer) [reply tests/js/realm-tests.js:2036]: It's tested twice, so I removed it from there.

File: tests/js/session-tests.js (hunk 5 above)
Comment (takameyer) [reply tests/js/session-tests.js:1189]: It breaks the test.  We can't copy a synced realm without a valid user session

File: types/app.d.ts (hunk 7 above)
Comment (kraenhansen) [reply types/app.d.ts:216]: Ahh - sorry :D I expanded and saw the entire file 🤦

File: tests/js/realm-tests.js (hunk 4 above)
Comment (kraenhansen) [reply tests/js/realm-tests.js:2036]: Ahh okay 👍

File: tests/js/session-tests.js (hunk 5 above)
Comment (kraenhansen) [reply tests/js/session-tests.js:1189]: Any idea (if /) why that worked before?

Comment (takameyer) [reply tests/js/session-tests.js]: Ah!  I can convert it to a local realm after user logout.  I'll test this

File: tests/js/session-tests.js [hunk 8]
```diff
@@ -1226,7 +1242,7 @@ module.exports = {
         _sessionStopPolicy: "immediately", // Make it safe to delete files after realm.close()
       },
       schema: [schemas.PersonForSync, schemas.DogForSync],
-      path: realm2Path,
+      path: `${realm1Path}copy2.realm`,
```
Comment (takameyer) [inline tests/js/session-tests.js:1245]: put duplicated string in a variable
